### PR TITLE
Added future_frame_data, to avoid blocking the VR render thread

### PR DESCRIPTION
**Pull request by asajeffrey**
At the moment, the VR render thread blocks on the VR device, waiting for the next frame. This PR makes it possible for the VR display to return a future, so the JS thread end up blocking rather than the VR render thread.

### CODE REVIEW

**Comment by asajeffrey** [comment]
Fixes #60

**Comment by asajeffrey** [comment]
https://github.com/servo/servo/pull/22938 is the matching servo PR.

File: rust-webvr-api/src/vr_future_frame_data.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    pub fn blocked() -> (VRResolveFrameData, VRFutureFrameData) {
+        let (send, recv) = channel();
+        (
+            VRResolveFrameData(State::Blocked(send)),
+            VRFutureFrameData(State::Blocked(recv)),
+        )
+    }
+
+    pub fn block(&mut self) -> VRFrameData {
```
Comment (paulrouget) [inline rust-webvr-api/src/vr_future_frame_data.rs]: Could you consume Self in that case? Any reason to keep the result around?

**Comment by paulrouget** [comment]
Looks good to me.

**Pull request merged**